keras_contrib/applications/densenet.py

```diff
@@ -382,10 +382,10 @@ def __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=False, dropou
     x_list = [x]
 
     for i in range(nb_layers):
-        x = __conv_block(x, growth_rate, bottleneck, dropout_rate, weight_decay)
-        x_list.append(x)
+        conv_block = __conv_block(x, growth_rate, bottleneck, dropout_rate, weight_decay)
+        x_list.append(conv_block)
 
-        x = concatenate(x_list, axis=concat_axis)
+        x = concatenate([x, conv_block], axis=concat_axis)
```

Comment:
Sorry for the late reply. 

Without the return of concat list, the upsample path will not be correct anymore. As in, the upsample block concatenates a long term dependency, and that is provided by the concat_list.

I tried modifying the code to work without that, but there are fewer number of parameters than the original paper and the current implementation, and that suggests that it may suffer during training.